shimmingtoolbox/cli/mask.py

```diff
@@ -128,3 +130,153 @@ def threshold(fname_input, output, thr):
+              help="Verbose: 0 = nothing, 1 = classic, 2 = expended.")
+# Options for _get_centerline
+# @click.option('--method', type=click.Choice(['optic', 'fitseg']), default='optic',
+#               help="(str): Method used for extracting the centerline: "
+#                    "- optic: automatic spinal cord detection method"
+#                    "- fitseg: fit a regularized centerline on an already-existing cord segmentation. It will "
+#                    "interpolate if slices are missing and extrapolate beyond the segmentation boundaries (i.e., "
+#                    "every axial slice will exhibit a centerline pixel). (default: optic)")
+# @click.option('--centerline_algo', type=click.Choice(['polyfit', 'bspline', 'linear', 'nurbs']), default='bspline',
+#               help="(str): Algorithm for centerline fitting. Only relevant with -method fitseg (default: bspline)")
+# @click.option('--centerline_smooth', default=30, help="(int): Degree of smoothing for centerline fitting. Only for "
+#                                                      "-centerline-algo {bspline, linear}. (default: 30)")
+def sct(fname_input, fname_output, contrast, centerline, file_centerline, brain, kernel, size, shape, remove_tmp,
+        verbose):
+
+    # Make sure input path exists
+    if not os.path.exists(fname_input):
+        raise RuntimeError("Input file does not exist")
+
+    # Get the number of dimensions
+    nii_input = nib.load(fname_input)
+    ndim = nii_input.ndim
+    # If 4d, last dimension is time, average last dim for better SNT
+    if ndim == 4:
+        input_3d = np.mean(nii_input.get_fdata(), 3)
+        nii_3d = nib.Nifti1Image(input_3d, affine=nii_input.affine, header=nii_input.header)
+        fname_mean = os.path.join(os.path.dirname(fname_output), 'mean_3d.nii.gz')
+        nib.save(nii_3d, fname_mean)
+        fname_process = fname_mean
+    # If not then only set the processing filename
+    else:
+        fname_process = fname_input
+
+    fname_seg = os.path.join(os.path.dirname(fname_output), 'seg.nii.gz')
+
+    # sct_get_centerline is faster than sct_deepseg_sc, however, it is a bit less accurate. More investigations needed
+    # in the future, this code is commented out so that we can persue investigation.
+    # # Get the centerline
+    # _get_centerline(fname_process, fname_seg)
+
+    # Run sct_deepseg_sc
+    # Use sct parameter convention
+    if remove_tmp:
+        remove = 1
+    else:
+        remove = 0
+
+    cmd = f"sct_deepseg_sc -i {fname_process} -o {fname_seg} -c {contrast} -centerline {centerline} -kernel {kernel} " \
+          f"-r {str(remove)} -v {str(verbose)}"
+    if centerline == 'file':
+        cmd += f" -file_centerline {file_centerline}"
+    if brain is not None and centerline == 'cnn':
+        cmd += f" -brain {brain}"
+
+    run_subprocess(cmd)
+
+    # Create the mask
+    run_subprocess(f"sct_create_mask -i {fname_process} -p centerline,{fname_seg} -size {size} -f {shape} "
```

Comment:
fname_seg here needs to be replaced with the output file mentioned in my comment above.

Reply:
See above comment